Tests/StreamChatTests/Repositories/AuthenticationRepository_Tests.swift

```diff
@@ -337,14 +646,17 @@ final class AuthenticationRepository_Tests: XCTestCase {
 
         let receivedError = try refreshTokenAndWaitForResponse(mockedError: mockedError)
 
-        XCTAssertCall("nextRetryDelay()", on: retryStrategy)
-        XCTAssertNotCall("resetConsecutiveFailures()", on: retryStrategy)
+        XCTAssertCall(RetryStrategy_Spy.Signature.nextRetryDelay, on: retryStrategy)
+        XCTAssertNotCall(RetryStrategy_Spy.Signature.resetConsecutiveFailures, on: retryStrategy)
         XCTAssertEqual(receivedError, mockedError)
     }
 
     func test_refreshToken_multipleCalls_theoreticApproach() throws {
-        try setTokenProvider(mockedResult: .success(.unique()))
-        retryStrategy.mock_nextRetryDelay.returns(0.1)
+        // Adding delay otherwise all the execution is 100% sync and we cannot simulate the scenario
+        try setTokenProvider(mockedResult: .success(.unique()), delay: .milliseconds(10))
```

Comment:
Theoretic approach: Calling refresh token 3 times on the same thread, without any delay
Realistic approach: Calling refresh token 3 times. The 2nd and 3rd times are called in the following runloop

> The delay also feels like a code smell 🤔
Otherwise things happen synchronously, and that would not mimic the reality of these tests.

Reply:
Probably worth adding a comment explaining this